src/prif.f90

```diff
@@ -53,8 +53,11 @@ module prif
     type(c_ptr) :: gex_team
     type(c_ptr) :: heap_mspace
     integer(c_intptr_t) :: heap_start
+    integer(c_size_t) :: heap_size
     type(prif_team_type), pointer :: parent_team
     type(handle_data), pointer :: coarrays
+    type(prif_coarray_handle) :: child_team_handle
+    type(child_team_info), pointer :: child_heap_info ! offset and size
```

Comment:
As explained in #123, I believe this representation of `prif_team_type` is a fundamentally incorrect implementation of `ISO_FORTRAN_ENV::TEAM_TYPE`, because it includes mutable team state unboxed inside the client-owned object. I expect this can lead to breakage as soon as the client makes a copy of a `TEAM_TYPE` aka `prif_team_type` value and then uses one copy to `CHANGE TEAM`.

I think we should go ahead and add the missing level of indirection requested in #123 ASAP, because the current declarations in prif.f90 mislead potential PRIF implementers into believing the current declarations represent a valid implementation strategy.

Reply:
I made this chance as part of a separate PR to get it in sooner, and then rebased this branch on to that.